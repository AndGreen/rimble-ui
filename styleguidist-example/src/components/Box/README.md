React component example:

```js
<Box bg="red">Push Me</Box>
```

```jsx
<Box
  p={5}
  fontSize={4}
  width={[ 1, 1, 1/2 ]}
  color='white'
  bg='salmon'>
  This is a Box
</Box>
```

You can add a custom props to an example wrapper:

```js { "props": { "className": "checks" } }
<Box>Iâ€™m transparent!</Box>
```

Or disable an editor by passing a `noeditor` modifier:

```jsx noeditor
<Box>Push Me</Box>
```

To render an example as highlighted source code add a `static` modifier:

```jsx static
import React from 'react';
```

Examples with all other languages are rendered only as highlighted source code, not an actual component:

```html
<Box size="large">Push Me</Box>
```

Any [Markdown](http://daringfireball.net/projects/markdown/) is **allowed** _here_.
