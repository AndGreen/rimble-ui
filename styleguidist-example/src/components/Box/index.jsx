import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Box } from 'rimble-ui'

// class BoxA extends Component {
//
//   render() {
//     return (<Box />);
//   }
//
// }

/** @component */
export default Box;
