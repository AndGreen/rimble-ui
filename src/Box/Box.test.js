import Box from './'

describe('Box component sanity', () => {
  it('has name', () => {
    expect(Box.displayName).toBe('Box')
  })

  it('matches default snapshot', () => {
    const component = render(<Box />)
    expect(component).toMatchSnapshot()
  })
})
