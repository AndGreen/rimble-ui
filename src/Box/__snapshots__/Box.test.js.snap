// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component sanity matches default snapshot 1`] = `
.c0 {
  text-align: left;
}

.c0 {
  box-sizing: border-box;
}

<div
  class="c0"
>
  
</div>
`;
